MQTT 5 Compilation error

I saw that MQTT 5 support has recently been added to contiki-ng (https://github.com/contiki-ng/contiki-ng/pull/1250). I am working on a project using TI CC2538 boards that would benefit from this, so I wanted to try it out. However, when I tried compiling the mqtt-client example with MQTT 5 enabled, I got these compilation errors (using arm-none-eabi-gcc version 9.2.1):

```
vagrant@ubuntu-bionic:~/iot/contiki-ng/examples/mqtt-client$ make
TARGET not defined, using target 'native'
  CC        ../../os/net/app-layer/mqtt/mqtt.c
../../os/net/app-layer/mqtt/mqtt.c: In function ‘write_out_props’:
../../os/net/app-layer/mqtt/mqtt.c:486:29: error: ‘prop’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
                             prop->val,
                             ^~~~
../../os/net/app-layer/mqtt/mqtt.c: In function ‘process_thread_mqtt_process’:
../../os/net/app-layer/mqtt/mqtt.c:582:5: error: ‘will_props’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
     write_out_props(pt, conn, will_props);
     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../os/net/app-layer/mqtt/mqtt.c:507:26: note: ‘will_props’ was declared here
   struct mqtt_prop_list *will_props = MQTT_PROP_LIST_NONE;
                          ^~~~~~~~~~
cc1: all warnings being treated as errors
../../Makefile.include:356: recipe for target 'build/native/obj/mqtt.o' failed
make: *** [build/native/obj/mqtt.o] Error 1
```

I have tried both `TARGET=cc2538dk` and target native (see above). To enable MQTT 5, I added the following line to `project-conf.h`:

```
#define MQTT_CONF_VERSION MQTT_PROTOCOL_VERSION_5
```

I realise that the MQTT 5 support is not part of an official release yet, but I had thought it should at least compile. Is this error perhaps caused by some misconfiguration on my part? How can I get it to compile?
Thanks for any assistance!

It now compiles for target native without error, but when I try compiling for cc2538dk, I get this:

```
vagrant@ubuntu-bionic:~/iot/contiki-ng/examples/mqtt-client$ make TARGET=cc2538dk
  CC        ../../arch/cpu/cc2538/dev/cc2538-rf.c
  CC        ../../arch/cpu/cc2538/./ieee-addr.c
  CC        ../../os/net/app-layer/mqtt/mqtt.c
  CC        ../../os/net/app-layer/mqtt/mqtt-prop.c
../../os/net/app-layer/mqtt/mqtt-prop.c: In function 'mqtt_get_next_in_prop':
../../os/net/app-layer/mqtt/mqtt-prop.c:508:30: error: converting a packed 'mqtt_vhdr_prop_t' {aka 'enum <anonymous>'} pointer (alignment 1) to a 'uint16_t' {aka 'short unsigned int'} pointer (alignment 2) may result in an unaligned pointer value [-Werror=address-of-packed-member]
  508 |                              NULL, NULL, (uint16_t *)prop_id);
      |                              ^~~~
In file included from ../../os/net/app-layer/mqtt/mqtt-prop.c:35:
../../os/net/app-layer/mqtt/mqtt.h:307:14: note: defined here
  307 | typedef enum {
      |              ^
cc1: all warnings being treated as errors
../../Makefile.include:356: recipe for target 'build/cc2538dk/obj/mqtt-prop.o' failed
make: *** [build/cc2538dk/obj/mqtt-prop.o] Error 1
```

--- a/os/net/app-layer/mqtt/mqtt.c
+++ b/os/net/app-layer/mqtt/mqtt.c
@@ -468,7 +468,7 @@ PT_THREAD(write_out_props(struct pt *pt, struct mqtt_connection *conn,
 {
   PT_BEGIN(pt);
 
-  struct mqtt_prop_out_property *prop;
+  static struct mqtt_prop_out_property *prop;
 
   if(prop_list) {
     DBG("MQTT - Writing %i property bytes\n", prop_list->properties_len + prop_list->properties_len_enc_bytes);
@@ -504,7 +504,7 @@ PT_THREAD(connect_pt(struct pt *pt, struct mqtt_connection *conn))
   PT_BEGIN(pt);
 
 #if MQTT_5
-  struct mqtt_prop_list *will_props = MQTT_PROP_LIST_NONE;
+  static struct mqtt_prop_list *will_props = MQTT_PROP_LIST_NONE;
   if(conn->will.properties) {
     will_props = (struct mqtt_prop_list *)list_head(conn->will.properties);
   }

--- a/tests/02-compile-arm-ports/Makefile
+++ b/tests/02-compile-arm-ports/Makefile
@@ -46,7 +46,7 @@ libs/trickle-library/zoul \
 lwm2m-ipso-objects/zoul:DEFINES=LWM2M_Q_MODE_CONF_ENABLED=1,LWM2M_Q_MODE_CONF_INCLUDE_DYNAMIC_ADAPTATION=1\
 lwm2m-ipso-objects/zoul:MAKE_WITH_DTLS=1 \
 mqtt-client/simplelink:BOARD=sensortag/cc2650:DEFINES=BOARD_CONF_SENSORS_DISABLE=1,TI_SPI_CONF_ENABLE=0 \
-mqtt-client/zoul:BOARD=firefly \
+mqtt-client/zoul:BOARD=firefly:DEFINES=MQTT_CONF_VERSION=5 \
 multicast/zoul \
 nullnet/zoul \
 platform-specific/cc2538-common/cc2538dk \

